shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/SchemaBuilder.java

```diff
@@ -64,42 +65,54 @@
     static {
         ShardingSphereServiceLoader.register(DialectTableMetaDataLoader.class);
     }
-    
-    /**
-     * Build ShardingSphere schema.
-     * 
-     * @param materials schema builder materials
-     * @return ShardingSphere schema
-     * @throws SQLException SQL exception
-     */
-    public static ShardingSphereSchema build(final SchemaBuilderMaterials materials) throws SQLException {
-        ShardingSphereSchema result = new ShardingSphereSchema();
```

Comment:
Hi do you think it is possible to modify the result of `build()` to `Map<ActualTableMetadata, LogicTableMetadata>`?
In that way, we can create a `ShardingSphereSchema` in `MetaDataContextsBuilder.java` with the list of `logicTables`. On the other hand, the list of `actualTables` is used to create `OptimizeContextFactory` in `MetaDataContextsBuilder.java`.

Reply:
Yes, this is fine. What I am puzzled is how to create the OptimizeContextFactory in standardmetadatacontexts? Many places use the standardmetadatacontexts construction method to create and use the related metaDataMap. By the way, I understand that ActualTableMetadata and LogicTableMetadata are similar in structure to ShardingSphereSchema?